GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/IncludedFolderTests.cs

```diff
@@ -0,0 +1,262 @@
+            string newFilePath = Path.Combine(newFolderPath, "test.txt");
+            File.WriteAllText(newFilePath, "New file content");
+            newFolderPath.ShouldBeADirectory(this.fileSystem);
+            newFilePath.ShouldBeAFile(this.fileSystem);
+            string[] fileSystemEntries = Directory.GetFileSystemEntries(newFolderPath);
+            fileSystemEntries.Length.ShouldEqual(33);
+
+            string projectedFolder = Path.Combine(newFolderPath, "Git");
+            projectedFolder.ShouldBeADirectory(this.fileSystem);
+            fileSystemEntries = Directory.GetFileSystemEntries(projectedFolder);
+            fileSystemEntries.Length.ShouldEqual(13);
+
+            string projectedFile = Path.Combine(newFolderPath, "ReturnCode.cs");
+            projectedFile.ShouldBeAFile(this.fileSystem);
+        }
+
+        [TestCase, Order(6)]
+        public void ReadFileThenChangingIncludeFoldersShouldRemoveFileAndFolder()
+        {
+            string fileToRead = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "RunFunctionalTests.bat");
+            this.fileSystem.ReadAllText(fileToRead);
+
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            string folderPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts");
+            folderPath.ShouldNotExistOnDisk(this.fileSystem);
+            fileToRead.ShouldNotExistOnDisk(this.fileSystem);
+        }
+
+        [TestCase, Order(7)]
+        public void CreateNewFileWillPreventRemoveIncludedFolder()
+        {
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder, "Scripts");
+            this.ValidateIncludedFolders(this.mainIncludedFolder, "Scripts");
+
+            string fileToCreate = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "newfile.txt");
+            this.fileSystem.WriteAllText(fileToCreate, "New Contents");
+
+            this.gvfsProcess.RemoveIncludedFolders("Scripts");
+            this.ValidateIncludedFolders(this.mainIncludedFolder, "Scripts");
+
+            string folderPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts");
+            folderPath.ShouldBeADirectory(this.fileSystem);
+            string[] fileSystemEntries = Directory.GetFileSystemEntries(folderPath);
+            fileSystemEntries.Length.ShouldEqual(6);
+            fileToCreate.ShouldBeAFile(this.fileSystem);
+
+            this.fileSystem.DeleteFile(fileToCreate);
+        }
+
+        [TestCase, Order(8)]
+        public void ModifiedFileShouldNotAllowIncludedFolderChange()
+        {
+            string modifiedPath = Path.Combine(this.Enlistment.RepoRoot, "Scripts", "RunFunctionalTests.bat");
+            this.fileSystem.WriteAllText(modifiedPath, "New Contents");
+
+            string output = this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            output.ShouldContain("Include was aborted");

```

Comment:
Also validate a non-zero return code (assuming that change is made to the verb)?